packages/edit-site/src/components/sidebar-edit-mode/page-panels/hooks.js

```diff
@@ -86,3 +92,79 @@ export function useCurrentTemplateSlug() {
 	return templates?.find( ( template ) => template.slug === entityTemplate )
 		?.slug;
 }
+
+// This is duplicated.
+const filterOutDuplicatesByName = ( currentItem, index, items ) =>
+	index === items.findIndex( ( item ) => currentItem.name === item.name );
+
+// Should we also get templates?
+export function useAvailablePatterns( template ) {
+	const currentThemeStylesheet = useSelect(
+		( select ) => select( coreStore ).getCurrentTheme().stylesheet
```

Comment:
The problem with moving this to a shared util is that it relies on the core store, so it can't be part of the block-editor package for example...